### PR TITLE
Editing the start delimeter of a raw-string-literal will update the end of the literal as well.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This happens in a proper transaction so that the user can hit undo to only have their actual edit apply.

### CODE REVIEW

File: src/EditorFeatures/CSharp/RawStringLiteral/RawStringLiteralCommandHandler_Return.cs [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Commanding;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Editor.Commanding.Commands;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.RawStringLiteral
+{
+    internal partial class RawStringLiteralCommandHandler : ICommandHandler<ReturnKeyCommandArgs>
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharp/RawStringLiteral/RawStringLiteralCommandHandler_Return.cs:22]: this is just a move of the previous logic into its own file.

File: src/EditorFeatures/CSharp/RawStringLiteral/RawStringLiteralCommandHandler.cs [hunk 2]
```diff
@@ -50,154 +36,5 @@ internal partial class RawStringLiteralCommandHandler : ICommandHandler<ReturnKe
         }
 
         public string DisplayName => CSharpEditorResources.Split_raw_string;
-
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharp/RawStringLiteral/RawStringLiteralCommandHandler.cs:53]: all this moved, unchanged to another partial type.

File: src/EditorFeatures/CSharp/RawStringLiteral/RawStringLiteralCommandHandler_Return.cs [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+    {
+        public CommandState GetCommandState(ReturnKeyCommandArgs args)
+            => CommandState.Unspecified;
+
+        /// <summary>
+        /// Checks to see if the user it typing <c>return</c> in <c>"""$$"""</c> and then properly indents the end
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharp/RawStringLiteral/RawStringLiteralCommandHandler_Return.cs:28]:
```suggestion
        /// Checks to see if the user is typing <c>return</c> in <c>"""$$"""</c> and then properly indents the end
```

**Review by allisonchou — APPROVED** [review]
lgtm! Awesome feature, I think this will definitely help the editing experience feel more polished 😄

**Pull request merged**